### PR TITLE
General election layout

### PR DESCRIPTION
**Pull request by NataliaLKB**
## What does this change?

This enables a new container for the snap election. **This code will be deleted after the election**, so we are less concerned with code quality.

cc @blongden73 @stephanfowler @guardian/dotcom-platform 

## What is the value of this and can you measure success?
A more impactful style for the fronts

## Does this affect other platforms - Amp, Apps, etc?
App will render this as a different default container, AMP will be unaffected. 

### Stage One:
**Desktop:**
image

**Tablet:**
image

**Mobile:**
image

### Stage Two:
**Desktop:**
image

**Tablet:**
image

**Mobile:**
image

## Tested in CODE?
Yes

### CODE REVIEW

File: common/app/slices/DynamicElection.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  override protected def optionalFirstSlice(stories: Seq[Story]): Option[(Slice, Seq[Story])] = {
+    val byGroup = segmentByGroup(stories)
+    val snap = byGroup.getOrElse(1, Seq.empty)
+
+    if (snap.nonEmpty) {
+      Some((FullMedia50, stories.drop(1)))
+    } else {
+      None
+    }
```
Comment (TBonnin) [inline common/app/slices/DynamicElection.scala]: Already mentioned in the `facial-tool` PR but you can achieve the same with
```
segmentByGroup(stories)
   .get(1)
   .filter(_.nonEmpty)
   .map(stories => (FullMedia50, stories.drop(1)))
```

**Review by TBonnin — APPROVED** [review]
lgtm

File: common/app/slices/DynamicElection.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    // optinal snap (1) - 2 - 4
+    stories.length match {
+      case 0 => Nil
+      case 1 => Nil
+      case 2 => Seq(ThreeQuarterTallQuarter1)
+      case 3 => Seq(ThreeQuarterTallQuarter1)
+      case 4 => Seq(ThreeQuarterTallQuarter1)
+      case 5 => Seq(ThreeQuarterTallQuarter1)
+      case 6 => Seq(ThreeQuarterTallQuarter1, QuarterQuarterQuarterQuarter)
+      case _ => Seq(ThreeQuarterTallQuarter1, QuarterQuarterQuarterQuarter)
```
Comment (jfsoul) [inline common/app/slices/DynamicElection.scala]: `case 2 | 3 | 4 | 5 => Seq(ThreeQuarterTallQuarter1)` or
` case l if (l >= 2 && l <= 5) => Seq(ThreeQuarterTallQuarter1)` or 
`case l if 2 until 5 contains l => Seq(ThreeQuarterTallQuarter1)`

(if only for future reference)

**Review by DiegoVazquezNanini — APPROVED** [review]

**Comment by NataliaLKB** [comment]
@TBonnin @jfsoul I am not ignoring your comments, but I am also not going to change anything related to code style/readability because this code will only be around until after the election.

I will merge this in the morning so I have the whole day to confirm nothing went wrong.

**Comment by paperboyo** [comment]
In Stage one tablet keeps the transparency, in Stege two - it doesn't? Cool, let's hope interactive can match colours.

**Pull request merged**